frontend/src/core/edit-app.tsx

```diff
@@ -119,7 +120,8 @@ export const EditApp: React.FC<AppProps> = ({ userConfig, appConfig }) => {
       .then(() => {
         setFilename(name);
         // Set document title: app_title takes precedence, then filename, then default
-        document.title = appConfig.app_title || name || "Untitled Notebook";
+        document.title =
+          appConfig.app_title || filetitle || "Untitled Notebook";
```

Comment:
you could do `Paths.basename` - this should handle windows too:

`appConfig.app_title || Paths.basename(name) || "Untitled Notebook";`

Reply:
Got it–ignore my recent commit.